### PR TITLE
Support circular references in ASDF-serialized models

### PR DESCRIPTION
**Pull request by eslavich**
### Description

This pull request updates the minimum asdf version to 2.6.0, and uses a new asdf feature (generators returned from `ExtensionType.from_tree_tagged` methods) to enable handling of models with circular user inverses.

I'm also removing explicit calls to `asdf.yamlutil.custom_tree_to_tagged_tree` and `asdf.yamlutil.custom_tree_to_tagged_tree` as those are no longer necessary.

### CODE REVIEW

**Comment by bsipocz** [comment]
note to reviewer: check the travis status on the second to last commit

**Comment by eslavich** [comment]
The last batch of failed checks were due to the GitHub outage, which seems to have been resolved.  Is it possible to re-run them?

**Comment by pllim** [comment]
Restarted CI

File: CHANGES.rst [hunk 1]
```diff
@@ -35,6 +35,9 @@ astropy.io.misc
 ^^^^^^^^^^^^^^^
 - Added serialization of parameter constraints fixed and bounds.  [#10082]
 
+- Fix ASDF serialization of circular model inverses, and remove explicit calls
+  to ``asdf.yamlutil`` functions that because unnecssary in asdf 2.6.0. [#10189]
```
Comment (nden) [inline CHANGES.rst:39]: `because they are unnecessary` ?

File: astropy/io/misc/asdf/tags/transform/compound.py [hunk 2]
```diff
@@ -139,8 +120,7 @@ def from_tree_transform(cls, node, ctx):
                 new_mapping.append(entry)
             else:
                 new_mapping.append(i)
-                transform = transform & ConstantType.from_tree(
-                    {'value': int(entry.value)}, ctx)
+                transform = transform & Const1D(int(entry.value))
```
Comment (nden) [inline astropy/io/misc/asdf/tags/transform/compound.py:123]: Casting to int is not necessary here. I'm guessing this was the idea for the old Const model but in modeling all parameters are turned into floats.

File: CHANGES.rst (hunk 1 above)
Comment (eslavich) [reply CHANGES.rst:39]: Oops, thanks, I intended to write "that became unnecessary" but went off the rails.

**Pull request merged**